client/driver/logging/syslog_server_unix_test.go

```diff
@@ -0,0 +1,46 @@
+package logging
+
+import (
+	"io/ioutil"
+	"net"
+	"os"
+	"testing"
+	"time"
+)
+
+const file = "/syslog_server.sock"
+
+func TestSyslogServer_Start_Shutdown(t *testing.T) {
+	path, err := ioutil.TempDir("", "")
+	if err != nil {
+		t.Fatalf("Failed to create temporary direcotry: %v", err)
+	}
+
+	path = path + file
+	defer os.Remove(path)
+
+	l, err := net.Listen("unix", path)
+	if err != nil {
+		t.Fatalf("Failed to listen unix socket: %v", err)
+	}
+
+	s := NewSyslogServer(l, make(chan *SyslogMessage, 2048), nil)
+
+	go s.Start()
+	if s.done {
+		t.Fatalf("expected running SyslogServer, but not running")
+	}
+
+	time.Sleep(1000 * time.Millisecond)
+
+	conn, err := net.Dial("unix", path)
```

Comment:
Since we are dialing into the connection, can we also send a message and see if the server receives it? You could just read from the channel you are passing into the server. And we could fail the test if we don't receive anything on the channel after sending the message to the server in 100ms or something.